Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -319,19 +324,22 @@ public class ProgramBuilder {
     /// Adoption of variables from a different program.
     /// Required when copying instructions between program.
     ///
-    private var varMaps = [[Variable: Variable]]()
+    private var varMaps = [VariableMap<Variable>]()
+    private var typeMaps = [VariableMap<Type>]()
     
     /// Prepare for adoption of variables from the given program.
     ///
     /// This sets up a mapping for variables from the given program to the
     /// currently constructed one to avoid collision of variable names.
     public func beginAdoption(from program: Program) {
-        varMaps.append([Variable: Variable]())
+        varMaps.append(VariableMap<Variable>())
```

Comment:
Nit: I don't think you need `<Variable>` here since the compiler should be able to infer it.